go-controller/pkg/ovn/policy.go

```diff
@@ -666,15 +666,42 @@ func (oc *Controller) processLocalPodSelectorSetPods(policy *knet.NetworkPolicy,
 		}
 
 		// Get the logical port info
+		// If the LSP is Scheduled for removal, wait until it's either gone
+		// or re-added to ensure we have the up to date version in the cache
+		// this usually only occurs with stateful-sets
 		logicalPort := util.GetLogicalPortName(pod.Namespace, pod.Name)
 		portInfo, err := oc.logicalPortCache.get(logicalPort)
 		// pod is not yet handled
 		// no big deal, we'll get the update when it is.
 		if err != nil {
+			klog.Warning("Failed to get LSP for pod %v/%v for networkPolicy %v err: %v", pod.Namespace, pod.Name, policy.Name, err)
 			continue
 		}
 
-		// this pod is somehow already added to this policy, then skip
+		for i := 0; i < 5; i++ {
+			if portInfo.expires.IsZero() {
+				klog.V(5).Infof("Fresh LSP %v for network policy %v found in cache", portInfo.name, policy.Name)
+				break
+			}
+			// 24 is chosen because gomega.Eventually default timeout is 50ms
+			// goovn transactions take less than 50ms usually as well so np create
+			// should be done within a couple iterations
+			time.Sleep(24 * time.Millisecond)
+
+			// Re-fetch the LSP
+			// This should not fail, since we already checked if pod was handled
```

Comment:
Even though it _should_ not fail, if it does: we will panic during the next iteration since `portInfo` will be `nil`, so you'll need to handle that case.

Reply:
Ack I made a broken assumption here, we should exit if getting the port succeeded and then proceeded to fail